src/Servers/Kestrel/test/InMemory.FunctionalTests/HttpsConnectionMiddlewareTests.cs

```diff
@@ -716,6 +709,81 @@ void ConfigureListenOptions(ListenOptions listenOptions)
         await AssertConnectionResult(stream, true);
     }
 
+    [ConditionalFact]
+    public async Task ServerCertificateChainInExtraStore()
+    {
+        var streams = new List<SslStream>();
+        CertHelper.CleanupCertificates();
+        (var clientCertificate, var clientChain) = CertHelper.GenerateCertificates("SslStream_ClinetCertificate_SendsChain", serverCertificate: false);
+
+        using (var store = new X509Store(StoreName.CertificateAuthority, StoreLocation.CurrentUser))
+        {
+            // add chain certificate so we can construct chain since there is no way how to pass intermediates directly.
+            store.Open(OpenFlags.ReadWrite);
+            store.AddRange(clientChain);
+            store.Close();
+        }
+
+        using (var chain = new X509Chain())
+        {
+            chain.ChainPolicy.VerificationFlags = X509VerificationFlags.AllFlags;
+            chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
+            chain.ChainPolicy.DisableCertificateDownloads = false;
+            var chainStatus = chain.Build(clientCertificate);
+            // Verify we can construct full chain
+            if (chain.ChainElements.Count < clientChain.Count)
+            {
+                throw new InvalidOperationException($"chain cannot be built {chain.ChainElements.Count}");
```

Comment:
@bartonjs Hrm we seem to hit an error on helix with this test on mac OSX only, should I just skip?

System.InvalidOperationException : chain cannot be built 1

https://dev.azure.com/dnceng/public/_build/results?buildId=1929785&view=ms.vss-test-web.build-test-results-tab&runId=49884456&resultId=121131&paneView=debug

Reply:
Cert chains generally behave the same on macOS as other platforms.  You'll probably benefit from printing out what the chain looked like and what all chain status values turned up (at the chain overall, and at the individual elements)

Reply:
Here's what the long cert chain looks like on windows when it's passing in the debugger in a sample run:
```
▶ | [0] | {[Subject]   CN=A Revocation Test CA 2, O=ServerCertificateChainInExtraStore  [Issuer]   CN=A Revocation Test CA 1, O=ServerCertificateChainInExtraStore  [Serial Number]   00FCC2A15A96F406D0  [Not Before]   8/10/2022 1:09:16 AM  [Not After]   11/14/2022 12:03:16 AM  [Thumbprint]   B70A0E006EF6E77280D355054ED33285901EA315 } | System.Security.Cryptography.X509Certificates.X509Certificate2
▶ | [1] | {[Subject]   CN=A Revocation Test CA 1, O=ServerCertificateChainInExtraStore  [Issuer]   CN=A Revocation Test CA 0, O=ServerCertificateChainInExtraStore  [Serial Number]   00F11BE02C645700DD  [Not Before]   8/10/2022 1:08:16 AM  [Not After]   11/14/2022 12:04:16 AM  [Thumbprint]   7EC341DBC3D9AE8A8E5F372C79440DA446C50616 } | System.Security.Cryptography.X509Certificates.X509Certificate2
▶ | [2] | {[Subject]   CN=A Revocation Test CA 0, O=ServerCertificateChainInExtraStore  [Issuer]   CN=A Revocation Test Root, O=ServerCertificateChainInExtraStore  [Serial Number]   00F2E046173643F2E2  [Not Before]   8/10/2022 1:07:16 AM  [Not After]   11/14/2022 12:05:16 AM  [Thumbprint]   5C2B53E144C96402F1ECA65A1CBBE77B234C6CCD } | System.Security.Cryptography.X509Certificates.X509Certificate2
▶ | [3] | {[Subject]   CN=A Revocation Test Root, O=ServerCertificateChainInExtraStore  [Issuer]   CN=A Revocation Test Root, O=ServerCertificateChainInExtraStore  [Serial Number]   3901199FF26B8536  [Not Before]   8/10/2022 1:06:16 AM  [Not After]   11/14/2022 12:06:16 AM  [Thumbprint]   649BAB9F7B5BBABEA7394373448360E7F686A338 } | System.Security.Cryptography.X509Certificates.X509Certificate2
```

Its actually failing now on Windows and OSX, but passes on Ubuntu??

Windows failure:
```
System.InvalidOperationException : chain cannot be built ChainStatus: A certificate chain could not be built to a trusted root authority.
Subject: CN=ServerCertificateChainInExtraStore, O=ServerCertificateChainInExtraStore Issuer: CN=A Revocation Test CA 2, O=ServerCertificateChainInExtraStore Status:
```

OSX failure
```
System.InvalidOperationException : chain cannot be built ChainStatus: One or more certificates required to validate this certificate cannot be found.
Subject: CN=ServerCertificateChainInExtraStore, O=ServerCertificateChainInExtraStore Issuer: CN=A Revocation Test CA 2, O=ServerCertificateChainInExtraStore Status: One or more certificates required to validate this certificate cannot be found.
111

Reply:
Filed https://github.com/dotnet/aspnetcore/issues/43273 to track figuring out what's going on with OSX.  Simplified the test to just check that any certs were added to the extra store, since I don't think runtime already has tests covering the lower level apis anyways so we don't need to the test code that verifies that the chain can be built successfully